### DISCUSSION TITLE
WPS201 Found module with too many imports: 13 > 12

### DISCUSSION THREAD

**Discussion by cidrblock** [ansible/ansible-navigator#701] [Linters, linting, formatting, pre-commit, CI]
In this case, majority of the imports are `from typing` (`actions/_actions`).

Is it possible to disregard the `from typing` imports?  Or what might be another approach?

```

import functools
import importlib
import logging
import re
from collections import namedtuple

from typing import Any
from typing import Callable
from typing import Dict
from typing import Generator
from typing import List
from typing import Tuple
```

**Comment by ssbarnea** [comment]
AFAIK, isort+black can take care of multiple imports from same module and we can easily ignore `WPS201`.

**Reply by webknjaz** [reply]
> AFAIK, isort+black can take care of multiple imports from same module and we can easily ignore `WPS201`.

This is not about that. isort just formats the imports, formatted is not the same as "not too much". This has the same problem as black — it doesn't have enough context to be useful in addressing this problem. Although, even if it did, it wouldn't be fixing this specific violation.

**Comment by webknjaz** [comment]
This violation is related to the module complexity. Hiding complexity is not a solution.

I think in such discussions, we need to copy-paste, or at least link the upstream reasoning/explanation. Copy from https://wemake-python-stylegui.de/en/latest/pages/usage/violations/complexity.html#wemake_python_styleguide.violations.complexity.TooManyImportsViolation.
Only with that, we'd be able to compare it with the context we're seeing.

Although we could think about a solution similar to what the recent ansible-core typing proposal suggests — they want to agree on doing `import typing as t` and use it as `t.List` / `t.Any` etc.
This suggests following a common coding standard and may be useful because often bare imports don't provide the right context within the given layer of abstraction. One exposed name in the imported module could be called differently in another module that imports that. One of the solutions could be using `from <other_context> import <other_name> as <proper_name>`. Another one is shifting the focus like `import <other_context>` followed by the use of `<other_context>.<proper_name>` in the module below.

**Reply by itisnotyourenv** [reply]
The domain is not active btw
